File: leetcode/1657.确定两个字符串是否接近.go
func closeStrings(word1 string, word2 string) bool {
    // 只需要判断字符出现的频率是否一样就可以
    m, n := len(word1), len(word2)
    if m != n {
        return false
    }
    arr1 := [26]int{}
    arr2 := [26]int{}
    mask1 := 0
    mask2 := 0
    for i := 0; i < n; i++ {
        arr1[word1[i]-'a'] += 1
        mask1 |= (1 << (word1[i]-'a'))
        arr2[word2[i]-'a'] += 1
        mask2 |= (1 << (word2[i]-'a'))
    }
    sort.Ints(arr1[:])
    sort.Ints(arr2[:])
    if mask1 != mask2 {
        return false
    }
    // Compare the frequency arrays
    for i := 0; i < 26; i++ {
        if arr1[i] != arr2[i] {
            return false
        }
    }

    return true
}

File: leetcode/1657.确定两个字符串是否接近.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        #凡是能够交换现有字符，都可以认为是可以进行排序的
        m,n=len(word1),len(word2)
        if m!=n:return False
        #种类一样，数量一样即可
        w_1=Counter(word1)
        w_2=Counter(word2)
        a=sorted(list(w_1.keys()))
        b=sorted(list(w_2.keys()))
        if a!=b:return False
        c=sorted(list(w_1.values()))
        d=sorted(list(w_2.values()))
        if c!=d:return False
        return True
        